pkg/plugin/aggregation/aggregator.go

```diff
@@ -193,15 +194,22 @@ func (a *Aggregator) handleResult(result *plugin.Result) error {
 		a.resultEvents <- result
 	}()
 
+	// Get the file extension to save the result as, falling back on guessing it.
+	var extension string
+	if result.Extension != "" {
+		extension = result.Extension
+	} else {
+		extension = result.GuessExtension()
```

Comment:
We only know the extension if one was provided in the upload URI (ie. /api/v1/results/global/foo.json), I'm not introducing any concept of per-plugin defaults or anything.

In case a plugin doesn't include a `.` in the upload URI, we're using the old behavior of guessing `json` unless it's the e2e plugin.  This may be a bit over-defensive though.  Any suggestions on a better approach?